### PR TITLE
Remove Remind Me Later option from the FRT discoverability banner

### LINKED ISSUE #6321: Remove the snooze behavior from Fraud & Risk discoverability banner
### Description

In order to prevent any confusion by merchants, we're going to remove the "Remind me later" button from the discoverability banner and always show the "Dismiss" button – that will keep its current behavior of hiding the banner forever after being clicked.

### Additional context
- p1684169257049719/1683880868.493449-slack-C04109UAJH5

### PR DESCRIPTION
**Pull request by allie500**
Fixes #6321 

#### Changes proposed in this Pull Request

This PR removes the Remind Me Later option from the Fraud and Risk Tools discoverability banner, leaving only the 'Learn more' and 'Dismiss' buttons.

#### Testing instructions

* Check out this branch.
* Build the client: `npm run build:client`
* All tests should pass.
* Delete the old banner settings option from DB: `npm run wp option delete wcpay_frt_discover_banner_settings`
* If required delete the new option that this PR introduces (if you tested previously): `npm run wp option delete wcpay_frt_discover_banner_dismissed`

**Laptop test**

* Navigate to the WCPay Overview page: `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`
* You should see the FRT discoverability banner displayed at the top of the page with a 'Learn more' button and a 'Dismiss' button only.

Screenshot on 2023-07-04 at 11_20_31

* Check the banner settings option in the DB: `npm run wp option get wcpay_frt_discover_banner_dismissed` 
  * It should return an empty option value
* Navigate to any other page in the admin, after that page loads, navigate back to the WCPay Overview page. The banner should display as it did before. 
* Click the 'Learn more' button. It should direct you to the WCPay settings page as it did before.
* Navigate back to the WCPay Overview page and click the `Dismiss` button. The banner should disappear.
* Check the banner settings option in the DB: `npm run wp option get wcpay_frt_discover_banner_dismissed` 
  * It should return: `1`
* Navigate to any other page in the admin, after that page loads, navigate back to the WCPay Overview page. The banner should ***not*** render.

**Smartphone test**

* Activate Jurassic Tube for your local site: `npm run tube:start`.
* Delete banner settings option from DB: `npm run wp option delete wcpay_frt_discover_banner_settings`
* Visit your local site in your smartphone and login.
* Navigate to the WCPay Overview page: `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`
* You should see the FRT discoverability banner displayed at the top of the page with a 'Learn more' button and a 'Dismiss' button only.
* Check the banner settings option in the DB: `npm run wp option get wcpay_frt_discover_banner_dismissed` 
  * It should return an empty option value
* Navigate to any other page in the admin, after that page loads, navigate back to the WCPay Overview page. The banner should display as it did before. 
* Click the 'Learn more' button. It should direct you to the WCPay settings page as it did before.
* Navigate back to the WCPay Overview page and click the `Dismiss` button. The banner should disappear.
* Check the banner settings option in the DB: `npm run wp option get wcpay_frt_discover_banner_dismissed` 
  * It should return: `1`
* Navigate to any other page in the admin, after that page loads, navigate back to the WCPay Overview page. The banner should ***not*** render.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6664` or branch name `update/6321-remove-frt-discoverability-banner-remind-option` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/6321-remove-frt-discoverability-banner-remind-option) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: fa798eb8d5a2a6de97bc3fe8b8501f9098d64d6f
- Build time: 2023-07-05 13:28:13 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/fraud-risk-tools-banner/index.tsx [hunk 1]
```diff
@@ -27,29 +24,15 @@ const FRTDiscoverabilityBanner: React.FC = () => {
 		try {
 			return JSON.parse( frtDiscoverBannerSettings );
 		} catch ( e ) {
-			return { remindMeCount: 0, remindMeAt: null, dontShowAgain: false };
+			return { dontShowAgain: false };
```
Comment (eduardoumpierre) [inline client/components/fraud-risk-tools-banner/index.tsx:27]: Now that we are holding just a single piece of information, what do you think about changing `frtDiscoverBannerSettings` to `frtDiscoverBannerDismissed` that will hold a boolean?

In that way, we won't have to have all the `JSON.parse` and `JSON.stringify` logic.

File: client/components/fraud-risk-tools-banner/index.tsx [hunk 2]
```diff
@@ -27,29 +24,15 @@ const FRTDiscoverabilityBanner: React.FC = () => {
...
-				remindMeCount: prevSettings.remindMeCount + 1,
-				remindMeAt: nowTimestamp + 3 * TIME.DAY_IN_MS,
-			};
-		} );
-	};
+	const showBanner = ! settings.dontShowAgain;
 
 	const setDontShowAgain = () => {
-		setSettings( ( prevSettings ) => {
+		setSettings( () => {
```
Comment (eduardoumpierre) [inline client/components/fraud-risk-tools-banner/index.tsx]: You can replace the callback function syntax by:
```js
setSettings( { dontShowAgain: true } );
```

**Review by eduardoumpierre — APPROVED** [review]
The changes make sense and the testing instructions worked as expected. I left some non-blocking comments, so feel free to merge without addressing them.

LGTM! :shipit:

File: client/components/fraud-risk-tools-banner/index.tsx (hunk 1 above)
Comment (allie500) [reply client/components/fraud-risk-tools-banner/index.tsx:27]: This is a good idea. I've made the changes in [ab8055f](https://github.com/Automattic/woocommerce-payments/pull/6664/commits/ab8055f5a75869456562f006d840b4a310fe5402). Unit tests and my manual tests pass. Could you take a look and see if the code looks how you expect?

File: client/components/fraud-risk-tools-banner/index.tsx (hunk 2 above)
Comment (allie500) [reply client/components/fraud-risk-tools-banner/index.tsx]: Thanks for flagging this. It looks a bit different with my changes based on your other recommendation. Change made in [ab8055f](https://github.com/Automattic/woocommerce-payments/pull/6664/commits/ab8055f5a75869456562f006d840b4a310fe5402).

File: client/components/fraud-risk-tools-banner/index.tsx (hunk 1 above)
Comment (allie500) [reply client/components/fraud-risk-tools-banner/index.tsx:27]: I made some minor changes in [c26b987](https://github.com/Automattic/woocommerce-payments/pull/6664/commits/c26b987677cc92ae7c8e9376741f9f8296abee0c).

File: includes/admin/class-wc-payments-admin.php [hunk 3]
```diff
@@ -780,7 +780,7 @@ private function get_js_settings(): array {
 				'isWelcomeTourDismissed' => WC_Payments_Features::is_fraud_protection_welcome_tour_dismissed(),
 			],
 			'accountDefaultCurrency'      => $this->account->get_account_default_currency(),
-			'frtDiscoverBannerSettings'   => get_option( 'wcpay_frt_discover_banner_settings', '' ),
+			'frtDiscoverBannerDismissed'  => get_option( 'wcpay_frt_discover_banner_dismissed' ),
```
Comment (eduardoumpierre) [inline includes/admin/class-wc-payments-admin.php]: I just realized that this change will make the discoverability banner visible for all merchants again. 😱

Maybe we should roll back to the last commit before the naming change?

**Review by eduardoumpierre — CHANGES_REQUESTED** [review]
Thanks for addressing the changes, Allie! After thinking more about this approach, I think it will cause the setting to be reset for all merchants due to the setting name change, so we should stick with the old nomenclature for now!

Sorry about the rework 😿

File: includes/admin/class-wc-payments-admin.php (hunk 3 above)
Comment (allie500) [reply includes/admin/class-wc-payments-admin.php]: That was a good catch! I'll re-work it tomorrow morning.

**Review by eduardoumpierre — APPROVED** [review]
The changes look good and everything worked as specified in the testing instructions.

Thanks for working on this.

LGTM! 🚢

**Pull request merged**